lib/matplotlib/backends/backend_pdf.py

```diff
@@ -429,7 +429,7 @@ def _flush(self):
             self.compressobj = None
 
 
-class PdfFile:
+class PdfFile(_backend_pdf_ps.CharacterTracker):
```

Comment:
I'm not quite convinced that this mixin style of `CharacterTracker` is easily understandable. You inject `_used_characters` which then other methods of the class use.

Would it be an alternative to add a `CharacterTracker` instance as an attribute?

Reply:
Agreed this may be architected better, will think about it.

Reply:
done